sys-devel/llvm-common/llvm-common-14.0.6.ebuild

```diff
@@ -1,28 +1,59 @@
-# Copyright 1999-2022 Gentoo Authors
+# Copyright 1999-2023 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=8
 
-inherit llvm.org
+inherit elisp-common llvm.org
 
 DESCRIPTION="Common files shared between multiple slots of LLVM"
 HOMEPAGE="https://llvm.org/"
 
 LICENSE="Apache-2.0-with-LLVM-exceptions UoI-NCSA"
 SLOT="0"
 KEYWORDS="amd64 arm arm64 ~ppc ppc64 ~riscv sparc x86 ~amd64-linux ~ppc-macos ~x64-macos"
+IUSE="emacs"
 
 RDEPEND="
 	!sys-devel/llvm:0
 "
+BDEPEND="
+	emacs? ( >=app-editors/emacs-23.1:* )
+"
 
-LLVM_COMPONENTS=( llvm/utils/vim )
+LLVM_COMPONENTS=( llvm/utils )
 llvm.org_set_globals
 
+SITEFILE="50llvm-gentoo.el"
+BYTECOMPFLAGS="-L emacs"
+
+src_compile() {
+	default
+
+	if use emacs ; then
+		elisp-compile "${S}" emacs/llvm-mode.el
+		elisp-compile "${S}" emacs/tablegen-mode.el
```

Comment:
Sorry for noticing only now, but what is `"${S}"` doing here?

Reply:
Saw it somewhere else, just removing that works right?